float32 expected_x
float32 expected_y
float32 current_angle
